backend/icpy/services/image_reference_service.py

```
@@ -222,6 +230,144 @@ async def list_references(self) -> Dict[str, ImageReference]:
+        async with self._lock:
+            updated: Dict[str, Dict[str, Any]] = {}
+            for image_id, ref in self._references.items():
+                abs_path = ref.get('absolute_path') or ''
+                file_type = ref.get('file_type')
+                session_ids = ref.get('session_ids')
+                timestamp = ref.get('timestamp') or 0
+
+                file_missing = True
+                try:
+                    file_missing = not (abs_path and Path(abs_path).exists())
+                except Exception:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)